plugin.rb

```diff
@@ -53,22 +54,17 @@ module ::DiscoursePrometheus; end
   if Discourse.running_in_rack?
     Thread.new do
       begin
-        DiscoursePrometheus::Demon.start
+        DiscoursePrometheus::CollectorDemon.start
+        DiscoursePrometheus::GlobalReporterDemon.start
         while true
-          DiscoursePrometheus::Demon.ensure_running
+          DiscoursePrometheus::CollectorDemon.ensure_running
+          DiscoursePrometheus::GlobalReporterDemon.ensure_running
           sleep 5
         end
       rescue => e
-        STDERR.puts "Failed to initialize prometheus web server from pid: #{Process.pid} #{e}"
+        STDERR.puts "Failed to initialize prometheus demons from pid: #{Process.pid} #{e}"
       end
     end
-
-    DiscoursePrometheus::Reporter::Global.start($prometheus_client)
-
-    # in dev we may use puma and it runs in a single process
-    if Rails.env == "development"
-      DiscoursePrometheus::Reporter::Process.start($prometheus_client, :web)
-    end
```

Comment:
Cleaned this up because I don't think we really support devving in puma any more. Maybe we should remove puma support from core entirely?